packages/@aws-cdk/aws-kinesisfirehose-destinations/test/s3-bucket.test.ts

```diff
@@ -0,0 +1,137 @@
+        CloudWatchLoggingOptions: {
+          Enabled: true,
+          LogGroupName: anything(),
+          LogStreamName: anything(),
+        },
+        RoleARN: stack.resolve(destinationRole.roleArn),
+      },
+    });
+    expect(stack).toHaveResource('AWS::Logs::LogGroup');
+    expect(stack).toHaveResource('AWS::Logs::LogStream');
+  });
+
+  it('allows disabling logging', () => {
+    new firehose.DeliveryStream(stack, 'DeliveryStream', {
+      destinations: [new firehosedestinations.S3Bucket(bucket, {
+        logging: false,
+      })],
+    });
+
+    expect(stack).toHaveResourceLike('AWS::KinesisFirehose::DeliveryStream', {
+      ExtendedS3DestinationConfiguration: {
+        CloudWatchLoggingOptions: ABSENT,
+      },
+    });
+  });
+
+  it('allows providing a log group', () => {
+    const logGroup = logs.LogGroup.fromLogGroupName(stack, 'Log Group', 'evergreen');
+
+    new firehose.DeliveryStream(stack, 'DeliveryStream', {
+      destinations: [new firehosedestinations.S3Bucket(bucket, {
+        logGroup,
+      })],
+    });
+
+    expect(stack).toHaveResourceLike('AWS::KinesisFirehose::DeliveryStream', {
+      ExtendedS3DestinationConfiguration: {
+        CloudWatchLoggingOptions: {
+          LogGroupName: 'evergreen',
+        },
+      },
+    });
+  });
+
+  it('creates a role when none is provided', () => {
+    const capturedRoleArn = Capture.aString();
+
+    new firehose.DeliveryStream(stack, 'DeliveryStream', {
+      destinations: [new firehosedestinations.S3Bucket(bucket)],
+    });
+
+    expect(stack).toHaveResourceLike('AWS::KinesisFirehose::DeliveryStream', {
+      ExtendedS3DestinationConfiguration: {
+        RoleARN: {
+          'Fn::GetAtt': [
+            capturedRoleArn.capture(),
+            'Arn',
+          ],
+        },
+      },
+    });
+    expect(stack).toMatchTemplate({
+      [capturedRoleArn.capturedValue]: {
+        Type: 'AWS::IAM::Role',
+      },
+    }, MatchStyle.SUPERSET);
+  });
+
+  it('grants read/write access to the bucket', () => {
+    const destination = new firehosedestinations.S3Bucket(bucket, { role: destinationRole });
+
+    new firehose.DeliveryStream(stack, 'DeliveryStream', {
+      destinations: [destination],
+    });
+
+    expect(stack).toHaveResourceLike('AWS::IAM::Policy', {
+      Roles: [stack.resolve(destinationRole.roleName)],
+      PolicyDocument: {
+        Statement: [
+          {
+            Action: [
+              's3:GetObject*',
+              's3:GetBucket*',
+              's3:List*',
+              's3:DeleteObject*',
+              's3:PutObject*',
+              's3:Abort*',
+            ],
+            Effect: 'Allow',
+            Resource: [
+              stack.resolve(bucket.bucketArn),
+              {
+                'Fn::Join': [
+                  '',
+                  [
+                    stack.resolve(bucket.bucketArn),
+                    '/*',
+                  ],
+                ],
+              },
```

Comment:
Let's format this in a slightly different way to conserve the total amount of lines:

```suggestion
              {
                'Fn::Join': ['', [
                  stack.resolve(bucket.bucketArn),
                  '/*',
                ]],
              },
```